### ISSUE TITLE
Not playing next song

### ISSUE THREAD

**Issue by amitiwary999** [ryanhoo/StylishMusicPlayer#11]
Player is not able to play next song automatically.User have to  click next song to play it.Nice job.Thank you

**Comment by ryanhoo** [comment]
Check the play mode, there are four play mode available: single, loop, shuffle, **list**. Only the list mode will stop when the list is finished playing.

**Comment by amitiwary999** [comment]
I checked it already and on clicking next button  it is not playing next song.

**Comment by JerryMissTom** [comment]
yes, in any play mode, the playing song will start from the beginning if you click next or pre button

**Comment by JerryMissTom** [comment]
in class Player, the method playNext() or playLast() will go to method play(),  and in method play(), it get current song and play it ,not the last  song or next. so the pre and next button will not response.  
在Player类中，方法playNext() 和playLast()中获取了nextSong 和lastSong,但是，并没有传给调用的play()方法，而play()方法获取的是currentSong,，所以上一首和下一首按钮不起作用

**Comment by DroidPulkit** [comment]
Is there a way in which I can contribute to this code base specially this bug?

**Comment by ryanhoo** [comment]
@DroidPulkit You fork this project, fix this bug, launch a pull request.

**Comment by JerryMissTom** [comment]
I fix this bug in my way, it will only play last or next song in the **same** playlist. In class Play, I add some code like as follows:
![image](https://cloud.githubusercontent.com/assets/15966403/19340828/6519ea42-915d-11e6-9632-c78b60d7017e.png)
![image](https://cloud.githubusercontent.com/assets/15966403/19340837/712347ca-915d-11e6-9b9f-820574f6e890.png)

if you have better idea, please inform me

**Comment by nillbiplob** [comment]
But it wasn't working for "All" playlist. So I fixed the bug by adding following code...

At PlayList.java

   public static PlayList fromLocalPlaylist(String name, List<Song> songs, int numOfSongs, int playingIndex) {
        PlayList playList = new PlayList();
        playList.setName(name);
        playList.setSongs(songs);
        playList.setNumOfSongs(numOfSongs);
        playList.setPlayingIndex(playingIndex);

```
    return playList;
}
```

AllLocalMusicFragment.java

 mAdapter.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(int position) {
              //  Song song = mAdapter.getItem(position);
                //RxBus.getInstance().post(new PlaySongEvent(song));
                // previous code was responsible for single song play

```
            /*
            new code by Biplob to play full playlist from "All" list.
             */

            PlayList playList =PlayList.fromLocalPlaylist(getString(R.string.mp_local_files_segmented_all),mAdapter.getData(),mAdapter.getData().size(),position);


            Log.e("Song in init playlist", playList.getNumOfSongs()+" total");

            RxBus.getInstance().post(new PlayListNowEvent(playList, position));

        }
    });
```
